src/component/admin/cors/CorsForm.test.tsx

```diff
@@ -0,0 +1,23 @@
+import {
+    parseInputValue,
+    formatInputValue,
+} from 'component/admin/cors/CorsForm';
+
+test('parseInputValue', () => {
+    const fn = parseInputValue;
+    expect(fn('')).toEqual([]);
+    expect(fn('a')).toEqual(['a']);
+    expect(fn('a\nb,,c,d,')).toEqual(['a', 'b', 'c', 'd']);
+    expect(fn('http://localhost:8080')).toEqual(['http://localhost:8080']);
+    expect(fn('https://example.com')).toEqual(['https://example.com']);
+    expect(fn('https://example.com/')).toEqual(['https://example.com']);
+    expect(fn('https://example.com/')).toEqual(['https://example.com']);
```

Comment:
is asterisk subdomain a valid option? `*.example.com`?

Reply:
Nope, for CORS we have to use `*` or a specific origin, as far as I can tell.